src/main/org/apache/tools/ant/listener/MailLogger.java

```diff
@@ -102,7 +102,7 @@
     private static final String DEFAULT_MIME_TYPE = "text/plain";
 
     /** Buffer in which the message is constructed prior to sending */
-    private StringBuffer buffer = new StringBuffer();
+    private StringBuilder buffer = new StringBuilder();
```

Comment:
I'm not too sure this change here, in this class is correct. The `StringBuffer` is a thread-safe class whereas the `StringBuilder` isn't. Having said that I haven't checked yet whether MailLogger class is expected to be thread safe nor have I checked its usage thoroughly.

Reply:
Given the `parallel` task and things like our execute framework that spawns new threads, loggers need to be thread safe, But to be honest I don't think we've ever verified `MailLogger` actually is.